tests/Unit/Controller/CourseControllerTest.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace Tests\Unit\Controller;
+
+use Tests\TestCase;
+use App\Models\Course;
+use App\Repositories\Course\CourseRepositoryInterface;
+use Mockery;
+use Illuminate\Database\Eloquent\Collection;
+use App\Http\Controllers\Admin\CourseController;
+use App\Http\Requests\CourseRequest;
+use Illuminate\Http\UploadedFile;
+use Faker\Factory as Faker;
+use Illuminate\Http\RedirectResponse;
+use Exception;
+
+class CourseControllerTest extends TestCase
+{
+    protected $courseMock;
+    protected $courseController;
+
+    public function setUp() : void
+    {
+        parent::setUp();
+        $this->courseMock = Mockery::mock(CourseRepositoryInterface::class);
+        $this->courseController = new CourseController($this->courseMock);
+    }
+
+    public function tearDown() : void
+    {
+        $this->courseController = null;
+        Mockery::close();
+        parent::tearDown();
+    }
+
+    public function test_index_return_view()
+    {
+        $this->courseMock
+            ->shouldReceive('getAll')
+            ->with(['image'], config('paginate.course_number'))
+            ->once()
+            ->andReturn(new Collection);
+        $result = $this->courseController->index();
+        $dataPassingToView = $result->getData();
+        $this->assertIsArray($dataPassingToView);
+        $this->assertArrayHasKey('courses', $dataPassingToView);
+        $this->assertEquals('admin.component.course', $result->getName());
+    }
```

Comment:
giải thích function này

Reply:
`$this->courseMock
            ->shouldReceive('getAll')
            ->with(['image'], config('paginate.course_number'))
            ->once()
            ->andReturn(new Collection);`